src/front-end/typescript/lib/pages/feedback.tsx

```diff
@@ -0,0 +1,158 @@
+
+export type Msg = GlobalComponentMsg<InnerMsg, Route>;
+
+export type RouteParams = null;
+
+const init: PageInit<RouteParams, SharedState, State, Msg> = async () => {
+  return {
+    loading: 0,
+    rating: undefined,
+    feedbackText: LongText.init({
+      id: 'feedback-text',
+      required: false,
+      placeholder: 'Describe your experience here.',
+      label: 'Please describe your experience.'
+    })
+  };
+};
+
+const startLoading: UpdateState<State> = makeStartLoading('loading');
+const stopLoading: UpdateState<State> = makeStopLoading('loading');
+
+const update: Update<State, Msg> = ({ state, msg }) => {
+  switch (msg.tag) {
+    case 'onChangeRating':
+      return [state.set('rating', msg.value)];
+    case 'onChangeFeedbackText':
+      return [updateField(state, 'feedbackText', msg.value)];
+    case 'validateFeedback':
+      return [validateField(state, 'feedbackText', validateFeedbackText)]
+    case 'submit':
+      state = startLoading(state);
+      return [
+        state,
+        async (state, dispatch) => {
+          const validatedRating = validateRating(state.rating);
+          if (validatedRating.tag === 'invalid') {
+            return state;
+          }
+          await api.createFeedback({
+            rating: validatedRating.value,
+            text: state.feedbackText.value
+          });
+
+          // Route to feedback submitted notice
+          dispatch(newRoute({
+            tag: 'notice' as 'notice',
+            value: {
+              noticeId: {
+                tag: 'feedbackSubmitted' as 'feedbackSubmitted',
+                value: undefined
+              }
+            }
+          }));
+          return stopLoading(state);
+        }
+      ];
+    default:
+      return [state];
+  }
+}
+
+const RatingSelector: ComponentView<State, Msg> = props => {
+  const { state, dispatch } = props;
+  const setRating = (value: Rating) => () => dispatch({ tag: 'onChangeRating', value });
+  const isGood = state.rating === 'good';
+  const isMeh = state.rating === 'meh';
+  const isBad = state.rating === 'bad';
+  return (
+    <Row className='mb-4'>
+      <Col xs='12' className='d-flex'>
+        <Icon className='mr-4' name='good-rating' color={isGood ? 'success' : 'secondary'} width={2.5} height={2.5} onClick={setRating('good')} />
+        <Icon className='mr-4' name='bad-rating' color={isBad ? 'danger' : 'secondary'} width={2.5} height={2.5} onClick={setRating('bad')} />
+        <Icon name='meh-rating' color={isMeh ? 'warning' : 'secondary'} width={2.5} height={2.5} onClick={setRating('meh')} />
+      </Col>
+    </Row>
+  );
+}
+
+const viewBottomBar: ComponentView<State, Msg> = props => {
+  const { state, dispatch } = props;
+  const validatedFeedbackText = validateGenericString(state.feedbackText.value, 'Description', 1, 2000);
+  const isLoading = state.loading > 0;
+  const isDisabled = isLoading || state.rating === undefined || validatedFeedbackText.tag === 'invalid';
```

Comment:
I suggest using `validateRating` here to follow the validation pattern used elsewhere in the code.